packages/react-switch/src/components/Switch/useSwitchState.ts

```diff
@@ -141,6 +138,10 @@ export const useSwitchState = (state: SwitchState) => {
     [onKeyDownCallback, setChecked],
   );
 
+  useEvent(eventData.element, 'pointermove', onPointerMove);
+  useEvent(eventData.element, 'pointerup', onPointerUp);
```

Comment:
The three hooks here all use the same element. You could encapsulate them all in one specific hook for the slider components e.g. `useSliderPointer` and create/cleanup all native DOM listeners in one single effect, which would be easier to maintain and have better encapsulation.